File: 00_Alternate_Languages/78_Sine_Wave/pascal/sinewave.pas
program sinewave;

procedure tabWriteLn(text: string; indent: integer);
begin
  Writeln(text:length(text)+indent);
end;

var
  a, t, b: integer;
begin
  tabWriteLn('SINE WAVE', 30);
  tabWriteLn('CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY', 15);
  Writeln();
  Writeln();
  Writeln();
  Writeln();
  Writeln();
  // REMARKABLE PROGRAM BY DAVID AHL
  b := 0;
  // START LONG LOOP
  for t := 0 to 40*4 do
  begin
    a := Trunc(26+25*Sin(t/4));
    if (b = 0) then
    begin
      tabWriteLn('CREATIVE', a);
      b := 1;
    end
    else
    begin
      tabWriteLn('COMPUTING', a);
      b := 0;
    end;
  end;
end.

File: 00_Alternate_Languages/78_Sine_Wave/C++/sinewave.cpp
#include <iostream> // std::cout, std::endl
#include <string>	// std::string(size_t n, char c)
#include <cmath>	// std::sin(double x)

int main()
{
	std::cout << std::string(30, ' ') << "SINE WAVE" << std::endl;
	std::cout << std::string(15, ' ') << "CREATIVE COMPUTING MORRISTOWN, NEW JERSEY" << std::endl;
	std::cout << std::string(5, '\n');

	bool b = true;

	for (double t = 0.0; t <= 40.0; t += 0.25)
	{
		int a = int(26 + 25 * std::sin(t));
		std::cout << std::string(a, ' ') << (b ? "CREATIVE" : "COMPUTING") << std::endl;
		b = !b;
	}

	return 0;
}
